lib/awful/placement.lua

```diff
@@ -434,6 +447,24 @@ local function area_remove(areas, elem)
     return areas
 end
 
+-- Convert 2 points into a rectangle
+local function rect_from_points(p1x, p1y, p2x, p2y)
+    return {
+        x      = p1x,
+        y      = p1y,
+        width  = p2x - p1x,
+        height = p2y - p1y,
+    }
+end
+
+-- Convert a rectangle and matrix info into a point
+local function rect_to_point(rect, corner_i, corner_j, n)
+    return {
+        x = rect.x + corner_i * math.floor(rect.width  / (n-1)),
+        y = rect.y + corner_j * math.floor(rect.height / (n-1)),
```

Comment:
Do you need the `n` argument? In the entire PR, only this commit adds a call to this function and this commit has a hardcoded `n=2` for that call. Then, `math.floor(rect.width / (2-1))` can be simplified to `rect.width`, I think.